sql/core/src/test/scala/org/apache/spark/sql/execution/python/PythonDataSourceSuite.scala

```diff
@@ -229,6 +229,38 @@ class PythonDataSourceSuite extends QueryTest with SharedSparkSession {
     }
   }
 
+  test("data source name conflicts") {
+    assume(shouldTestPandasUDFs)
+    val dataSourceScript =
```

Comment:
We won't actually need this script. We could create an empty UserDefinedPythonDataSource. I am adding a util for this `PythonUtils.createPythonFunction` at https://github.com/apache/spark/pull/44504. I can followup once that's merged.